src/main/java/nextstep/ladder/domain/Ladder.java

```diff
@@ -9,21 +10,56 @@ public class Ladder {
 
     private final List<Line> lines;
 
+    private final LadderResult ladderResult;
 
-    public Ladder(int height, int length, LineGenerator lineGenerator) {
-        assertValidDepth(height);
 
+    public Ladder(int height, int length, LineGenerator lineGenerator, List<String> results) {
         this.height = new LadderHeight(height);
+        this.ladderResult = new LadderResult(results, length + 1);
         this.lines = Stream.generate(() -> new Line(length, lineGenerator))
                 .limit(height)
                 .collect(Collectors.toUnmodifiableList());
     }
 
-    private void assertValidDepth(int height) {
-        String invalidRangeMessage = "[사다리] 사다리 높이는 1 이상이어야 합니다.";
-        if (height < 1) {
-            throw new IllegalArgumentException(invalidRangeMessage);
+    public GameResult getGameResult(Users users) {
+        List<String> orderLadderResult = matchLadderResultByOrder();
+        return new GameResult(orderLadderResult, users);
+    }
+
+    private List<String> matchLadderResultByOrder() {
+
+        List<String> orderLadderResult = new ArrayList<>();
+
+        for (int i = 0; i < ladderResult.getResults().size(); i++) {
+            orderLadderResult.add(ladderResult.getResults().get(findIndex(lines, i)));
+        }
+
+        return orderLadderResult;
+    }
+
+    private int findIndex(List<Line> lines, int startIndex) {
+        int nextIndex = startIndex;
+
+        for (Line line : lines) {
+            nextIndex = getNextIndex(line, nextIndex);
```

Comment:
`Line`에서 값을 꺼내서 처리하는 대신에, 메시지를 보내서 처리해보면 어떨까요?
```suggestion
            nextIndex = line.getNextIndex(nextIndex);
```